.github/workflows/pr.yaml

```diff
@@ -73,7 +76,55 @@ jobs:
 
       - name: Test examples
         run: yarn test:examples
-      
+
+  build-lint-test-example-widgets-with-sdk:
+    name: Build, Lint & Test Example Widgets with latest SDK
+    runs-on: ubuntu-latest-8-cores
+    env:
+      NODE_OPTIONS: --max-old-space-size=14366
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          fetch-depth: 0
+
+      - name: setup
+        uses: ./.github/actions/setup
+
+      - name: Setup playwright
+        uses: ./.github/actions/setup-playwright
+
+      - name: Get current SDK version string
+        uses: actions/github-script@v7.0.1
+        with: 
+          script: |
+            const { data: latestRelease } = await github.rest.repos.getLatestRelease({
+              owner: context.repo.owner,
+              repo: context.repo.repo
+            });
+            console.log("Current SDK version: ", latestRelease.tag_name);
+            return latestRelease.tag_name;
+          result-encoding: string
+        id: version
+
+      - name: Install SDK at current version
+        run: yarn workspace @examples/sdk-load-widgets-with-nextjs add @imtbl/sdk@${{ steps.version.outputs.result }}
+
+      - name: Build next app
+        run: yarn workspace @examples/sdk-load-widgets-with-nextjs build
+
+      - name: Build everything
+        run: yarn build
+
+      - name: Lint examples
+        run: yarn workspace @examples/sdk-load-widgets-with-nextjs lint
+
+      - name: Prepare widgets bundle for example app
+        run: yarn workspace @examples/sdk-load-widgets-with-nextjs prepare:widgets
+
+      - name: Test Widget example
+        run: yarn workspace @examples/sdk-load-widgets-with-nextjs test:latest
```

Comment:
The main things https://www.npmjs.com/package/@imtbl/sdk `build-lint-test-example-widgets-with-sdk` is trying to achieve are:

* build `@examples/sdk-load-widgets-with-nextjs` with the latest released production build (i.e what is `latest` here https://www.npmjs.com/package/@imtbl/sdk?activeTab=versions, current `1.56.0`)
* by default at runtime, the `checkoutSDK.widgets()` loads the latest compatible version of a remote JS widgets bundle (i.e if a user has SDK `1.55.0`, they will load Widgets from `1.56.0` at runtime)
* instead of the default behaviour, I want to load a local JS widgets bundle built from the branch that has triggered this PR
* then validate with an e2e test that SDK `1.56.0` works with local branch Widgets JS, so that when this branch Widgets gets released as `1.57.0`, we have some confidence that SDK `1.56.0` is compatible with Widgets `1.57.0`


That's why this is a separate job to `build-lint-test-examples`.
This example app is also run in that job, so we do test this example app against the current unreleased version of the SDK as well.

Reply:
Ah ok, that clears up that confusion. Cheers for the break down ❤️ 

I've suggested a simplification to this [here](https://github.com/immutable/ts-immutable-sdk/pull/2282#discussion_r1796267830) to clean it up a bit but other than that it should be ok.